aptos-move/block-executor/src/scheduler_v2.rs

```diff
@@ -66,3 +422,1400 @@ impl ExecutionQueueManager {
+    /// most up-to-date status of transactions.
+    fn propagate(&self, mut propagation_queue: BTreeSet<usize>) -> Result<(), PanicError> {
+        while let Some(task_idx) = propagation_queue.pop_first() {
+            // Make sure the conditions are checked under dependency lock.
+            let mut aborted_deps_guard = self.aborted_dependencies[task_idx].lock();
+
+            // checks the current status to determine whether to propagate add / remove stall,
+            // calling which only affects its currently not_stalled (or stalled) dependencies.
+            // Allows to store indices in propagation queue (not add or remove commands) & avoids
+            // handling corner cases such as merging commands (as propagation process is not atomic).
+            if self
+                .txn_statuses
+                .shortcut_executed_and_not_stalled(task_idx)
+            {
+                // Still makes sense to propagate remove_stall.
+                aborted_deps_guard.remove_stall(&self.txn_statuses, &mut propagation_queue)?;
+            } else {
+                // Not executed or stalled - still makes sense to propagate add_stall.
+                aborted_deps_guard.add_stall(&self.txn_statuses, &mut propagation_queue)?;
+            }
+        }
+        Ok(())
+    }
+
+    /// Initiates the abort process for a specific transaction incarnation via `ExecutionStatuses`.
+    ///
+    /// This is a wrapper around `[ExecutionStatuses::start_abort]`.
+    /// If the abort is successfully initiated (i.e., `[ExecutionStatuses::start_abort]` returns `Ok(true)`):
+    /// - Increments the `SPECULATIVE_ABORT_COUNT` counter.
+    /// - Clears any speculative transaction logs for the `txn_idx` (as its current execution
+    ///   attempt is being aborted).
+    ///
+    /// Returns the result of `[ExecutionStatuses::start_abort]`.
+    fn start_abort(&self, txn_idx: TxnIndex, incarnation: Incarnation) -> Result<bool, PanicError> {
+        let ret = self.txn_statuses.start_abort(txn_idx, incarnation)?;
+        if ret {
+            // Increment the counter and clear speculative logs (from the aborted execution).
+            counters::SPECULATIVE_ABORT_COUNT.inc();
+            clear_speculative_txn_logs(txn_idx as usize);
+        }
+
+        Ok(ret)
+    }
+
+    /// Initiates the execution of a transaction via `ExecutionStatuses`.
+    ///
+    /// This is a direct wrapper around `[ExecutionStatuses::start_executing]`.
+    /// It attempts to transition the transaction at `txn_idx` to the `Executing` state.
+    ///
+    /// Returns `Ok(Some(incarnation))` if successful, `Ok(None)` if the transaction
+    /// was not in a state to start execution (e.g., not `PendingScheduling`), or
+    /// `Err(PanicError)` if an error occurs within `ExecutionStatuses`.
+    fn start_executing(&self, txn_idx: TxnIndex) -> Result<Option<Incarnation>, PanicError> {
```

Comment:
Name is confusing. You only mark here, not really start the execution.

Reply:
you do start logical execution which contains VM execution and a bunch of post-processing as well before finish execution